sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -531,6 +531,7 @@ object FunctionRegistry {
     expression[ToNumber]("to_number"),
     expression[TryToNumber]("try_to_number"),
     expression[ToCharacter]("to_char"),
+    expression[ToCharacter]("to_varchar", setAlias = true, Some("3.4.0")),
```

Comment:
Change the version to 3.5.0. We are not going to backport this alias back to already released 3.4.0.